pkg/metrics/collector.go

```diff
@@ -18,17 +18,57 @@ package metrics
 
 import (
 	"context"
+	"fmt"
 
 	directpvtypes "github.com/minio/directpv/pkg/apis/directpv.min.io/types"
 	"github.com/minio/directpv/pkg/client"
 	"github.com/minio/directpv/pkg/consts"
+	"github.com/minio/directpv/pkg/device"
 	"github.com/minio/directpv/pkg/sys"
 	"github.com/minio/directpv/pkg/types"
+	"github.com/minio/directpv/pkg/utils"
 	"github.com/minio/directpv/pkg/xfs"
 	"github.com/prometheus/client_golang/prometheus"
 	"k8s.io/klog/v2"
 )
 
+type driveStats struct {
+	readSectorBytes  float64
+	readTicks        float64
+	writeSectorBytes float64
+	writeTicks       float64
+	timeInQueue      float64
+}
+
+func getDriveStats(driveName string) (*driveStats, error) {
+	stat, err := device.GetStat(driveName)
+	switch {
+	case err != nil:
+		return nil, err
+	case len(stat) == 0:
+		return nil, fmt.Errorf("unable to read stat from drive %v", driveName)
+	case len(stat) < 10:
+		return nil, fmt.Errorf("invalid stat format from drive %v", driveName)
+	}
+
+	hardwareSectorSize, err := device.GetHardwareSectorSize(driveName)
+	switch {
+	case err != nil:
+		return nil, err
+	case hardwareSectorSize == 0:
+		hardwareSectorSize = 512 // Use default value
+	}
+
+	// Refer https://www.kernel.org/doc/Documentation/block/stat.txt for meaning of each field.
+	return &driveStats{
+		readSectorBytes:  float64(stat[2] * hardwareSectorSize),
+		readTicks:        float64(stat[3]),
+		writeSectorBytes: float64(stat[6] * hardwareSectorSize),
+		writeTicks:       float64(stat[7]),
+		timeInQueue:      float64(stat[9]),
```

Comment:
```suggestion
		timeInQueue:      float64(stat[10]),
```

I see that timeInQueue is 11th value in the stat line as per https://www.kernel.org/doc/Documentation/block/stat.txt